### PR TITLE
Add ferns

### PR DESCRIPTION
**Pull request by e1ven**
#### Summary

SUMMARY: Content "Adds ferns and fiddleheads"
#### Purpose of change

Adds Ferns to forests and heavy forests.
Ferns can yield fiddleheads in the spring as a potential food source.

#### Describe the solution
I had initially tried this [as a mod](https://discourse.cataclysmdda.org/t/mod-that-adds-ferns/21533), but was advised that PRs are the preferred method for this project.

These Ferns can be harvested for fiddleheads in the spring.
Eating uncooked fiddleheads [can make you sick](https://honest-food.net/bracken-fern-edible/) - I tried to represent this with a `healthy: -2` modifier.

* There is a recipe for boiling fiddleheads - This should make them edible. 
If you boil them in some sort of stock, it can increase their calories.

* There is two recipes for sauteeing fiddleheads -
Since sauteed fiddleheads still requiring boiling first, there are two versions.
One takes boiled fiddleheads, the other will boil them as part of the process.
I set this to work with anything that seemed like it would work as a pan-frying fat.

I set `batch_time_factors` because once you have a pot of boiling water going, you can add more to it pretty easily.

#### Describe alternatives you've considered
I originally tested a fern as a furniture object rather than terrain. 
I think it works better to have it as terrain, since there isn't going to be grass underneath it.

#### Additional context

As an enhancement, a "Deluxe" recipe could be added which adds some garlic, salt or pepper.
They could also be added to other areas where a vegetable can be used, such as broth.
Ferns don't have seeds in the traditional sense - They use spores. In theory a process could be added to use sheets of paper to extract these.

### CODE REVIEW

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/mod-that-adds-ferns/21533/12

File: data/json/regional_map_settings.json [hunk 1]
```diff
@@ -267,6 +268,7 @@
             "clear_types": false,
             "types": {
               "t_underbrush": 30,
+              "t_fern": 15,
```
Comment (Malkeus) [inline data/json/regional_map_settings.json]: I'll have to try this out, but this number seems too high. Do ferns really make up 1/3 of the ground cover in a New England forest?

Comment (e1ven) [reply data/json/regional_map_settings.json]: I'm happy to change it to whatever feels the best for gameplay.
According to (Penn State)[https://ecosystems.psu.edu/research/centers/private-forests/news/2016/new-forest-science-fact-sheet-on-fern-impacts-and-control]

>The latest U.S. Forest Service forest inventory and analysis data indicate that rhizomatous ferns species, hay-scented, New York, and bracken fern, now comprise over 20% of Pennsylvania’s forest understories.

Comment (Malkeus) [reply data/json/regional_map_settings.json]: I added the mod and took out the looks-like field. Here's a picture of a forest at max zoom:
image
1 zoom level in from max:
image
Thinning them out a bit seems appropriate. I'll reduce my numbers to 9 and come back with a picture.

**Comment by Malkeus** [comment]
Results at 9, Max Zoom:
image
1 zoom in from max:
image
It's a subtle difference, but they don't seem quite as dominant as before. I'm not sure if this matches 20%, but it seems acceptable to me. Anyone else care to weigh in?

The only other complaint I could see coming from this is the "look-like" field is set to t_underbrush which will make those of us still using DeadPeople tileset suffer. But DeadPeople is discontinued so this will probably sort itself out if/when a new official tileset is made. Someone can push a PR to Chaosvolts DeadPeople backup repository if people get annoyed or fix it locally. 

I like it! Thanks for taking the time to push a PR. :)

File: data/json/items/comestibles/veggy_dishes.json [hunk 2]
```diff
@@ -728,5 +728,47 @@
...
+    "name": "boiled fiddleheads",
+    "name_plural": "boiled fiddleheads",
+    "weight": "50 g",
+    "flags": [ "FREEZERBURN", "EATEN_HOT" ],
+    "color": "light_green",
+    "spoils_in": "2 days 12 hours",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": 35,
+    "description": "Boiled Fiddleheads. Not bad, but would be better sauteed in butter.",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/veggy_dishes.json]:
```suggestion
    "description": "Boiled fiddleheads.  Not bad, but would be better sauteed in butter.",
```

File: data/json/items/comestibles/raw_veggy.json [hunk 3]
```diff
@@ -758,5 +758,27 @@
...
+    "name": "handful of fiddleheads",
+    "name_plural": "handfuls of fiddleheads",
+    "weight": "50 g",
+    "flags": [ "FREEZERBURN" ],
+    "color": "light_green",
+    "spoils_in": "2 days 12 hours",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": 18,
+    "description": "A handful of immature fern fronds, still curled up like the head of a fiddle. Delicious when cooked, but consuming raw can cause food poisoning.",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/raw_veggy.json]:
```suggestion
    "description": "A handful of immature fern fronds, still curled up like the head of a fiddle.  Delicious when cooked, but consuming raw can cause food poisoning.",
```

File: data/json/items/comestibles/veggy_dishes.json [hunk 4]
```diff
@@ -728,5 +728,47 @@
...
+    "name": "sauteed fiddleheads",
+    "name_plural": "sauteed fiddleheads",
+    "weight": "100 g",
+    "flags": [ "FREEZERBURN", "EATEN_HOT" ],
+    "color": "light_green",
+    "spoils_in": "2 days 12 hours",
+    "comestible_type": "FOOD",
+    "symbol": "%",
+    "calories": 160,
+    "description": "Fiddleheads sauteed in fat. Tender and Delicious.",
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/veggy_dishes.json]:
```suggestion
    "description": "Fiddleheads sauteed in fat.  Tender and delicious.",
```

**Comment by Rivet-the-Zombie** [comment]
>If fiddleheads are indeed poisonous in raw form, consider using "use_action": "POISON" in addition (or instead) of just adding negative healthy value.

They're mildly carcinogenic, but not sufficiently toxic to count as poisonous by DDA's standards. The negative health value for raw fiddleheads seems appropriate here.

File: data/json/items/comestibles/veggy_dishes.json (hunk 4 above)
Comment (e1ven) [reply data/json/items/comestibles/veggy_dishes.json]: Thanks. Applied.

**Pull request merged**